src/messages/manifestjson.js

```diff
@@ -21,6 +21,19 @@ export const MANIFEST_FIELD_INVALID = {
   file: MANIFEST_JSON,
 };
 
+export const MANIFEST_FIELD_PRIVILEGEDONLY = 'MANIFEST_FIELD_PRIVILEGEDONLY';
+export function manifestFieldPrivilegedOnly(fieldName) {
+  return {
+    code: MANIFEST_FIELD_PRIVILEGEDONLY,
+    message: i18n._(`"${fieldName}" is ignored for non-privileged add-ons.`),
+    description: i18n._(
+      oneLine`"${fieldName} manifest field is only used for privileged
+             and temporarily installed extensions.`
```

Comment:
the error ID suggests what most of the developer would actually care about (the fact that the field is only expected to be used by privileged addons), given that the description can be a bit longer I did also mention the temporarily installed addon (which may be useful info for the developers of privileged addons, in case they are wondering if the field was actually used while they were testing they addon locally as unsigned).

Having said that, I'm ok to remove the mention to the temporarily installed extensions if you strongly feel that would be better.

Reply:
```js
    message: i18n._(`"${fieldName}" is ignored for non-standard add-ons.`),
```

would fix the issue I guess. Mentioning temporary extensions sounds like a good idea to me.

Reply:
Personally I think that:
- it is not a big deal that the id of the rule is MANIFEST_FIELD_PRIVILEGEDONLY but the longer description also mention that it is also used on temporarily installed extensions.
- "non-standard add-ons" may be slightly less understandable from an extension developer perspective, I'm not sure we use that terminology anywhere else, and so I would personally prefer to leave it as "non-privileged add-ons"
- the fact that "message" is explicitly calling out only non-privileged doesn't sound like a big deal, in the end that is what we care more that the developer is aware of right away.

And so I didn't apply any further changed on this part.

As I did mention in my previous comment I'm ok to apply further changes if we feel that we should.